src/test/java/de/tum/in/www1/artemis/service/GitlabCIServiceTest.java

```diff
@@ -0,0 +1,138 @@
+        assertThat(health.isUp()).isTrue();
+        assertThat(health.getAdditionalInfo()).containsEntry("cf.", "Version Control Server").containsEntry("url", gitlabServerUrl);
+    }
+
+    @Test
+    @WithMockUser(roles = "INSTRUCTOR", username = "instructor1")
+    public void testGetWebHookUrl() {
+        // TODO
+        var result = continuousIntegrationService.getWebHookUrl(null, null);
+        assertThat(result).isEqualTo(Optional.empty());
+    }
+
+    @Test
+    @WithMockUser(roles = "INSTRUCTOR", username = "instructor1")
+    public void testGetBuildStatus() {
+        // TODO
+        var result = continuousIntegrationService.getBuildStatus(null);
+        assertThat(result).isNull();
+    }
+
+    @Test
+    @WithMockUser(roles = "INSTRUCTOR", username = "instructor1")
+    public void testTriggerBuildSuccess() throws GitLabApiException {
+        mockTriggerBuild(null);
+        ProgrammingExerciseStudentParticipation participation = new ProgrammingExerciseStudentParticipation();
+        participation.setRepositoryUrl("http://some.test.url/PROJECTNAME/REPONAME-exercise.git");
+        continuousIntegrationService.triggerBuild(participation);
+    }
+
+    @Test
+    @WithMockUser(roles = "INSTRUCTOR", username = "instructor1")
+    public void testTriggerBuildFails() throws GitLabApiException {
+        mockTriggerBuildFailed(null);
+        ProgrammingExerciseStudentParticipation participation = new ProgrammingExerciseStudentParticipation();
+        participation.setRepositoryUrl("http://some.test.url/PROJECTNAME/REPONAME-exercise.git");
+        assertThatThrownBy(() -> continuousIntegrationService.triggerBuild(participation)).isInstanceOf(GitLabCIException.class);
+    }
+
+    @Test
+    @WithMockUser(roles = "INSTRUCTOR", username = "instructor1")
+    public void testConfigureBuildPlanSuccess() throws Exception {
+        continuousIntegrationTestService.testConfigureBuildPlan();
+    }
+
+    @Test
+    @WithMockUser(roles = "INSTRUCTOR", username = "instructor1")
+    public void testConfigureBuildPlanFails() throws GitLabApiException {
+        mockAddBuildPlanToGitLabRepositoryConfiguration(true);
+
+        ProgrammingExerciseStudentParticipation participation = new ProgrammingExerciseStudentParticipation();
+        participation.setRepositoryUrl("http://some.test.url/PROJECTNAME/REPONAME-exercise.git");
+        assertThatThrownBy(() -> continuousIntegrationService.configureBuildPlan(participation, "main")).isInstanceOf(GitLabCIException.class);
+    }
+
+    @Test
+    @WithMockUser(roles = "INSTRUCTOR", username = "instructor1")
+    public void testCreateBuildPlanForExercise() throws GitLabApiException, URISyntaxException {
+        mockAddBuildPlanToGitLabRepositoryConfiguration(false);
+        continuousIntegrationService.createBuildPlanForExercise(null, null, new VcsRepositoryUrl("http://some.test.url/PROJECTNAME/REPONAME-exercise.git"), null, null);
```

Comment:
I’d have assumed that the mock is used after the call to assert that some method was called with the expected changes to the Gitlab project.